index.bs

```diff
@@ -645,6 +648,26 @@ Modify the <a spec=HTML>download the hyperlink</a> algorithm to ensure that down
 
 Many specifications need to be patched so that, if a given algorithm invoked in a [=prerendering browsing context=], most of its work is deferred until the browsing context is [=prerendering browsing context/activated=]. This is tricky to do uniformly, as many of these specifications do not have great hygeine around <a href="https://html.spec.whatwg.org/multipage/webappapis.html#event-loop-for-spec-authors">using the event loop</a>. Nevertheless, the following sections give our best attempt.
 
+<h4 id="delay-while-prerendering">The {{[DelayWhilePrerendering]}} extended attribute</h4>
+
+To abstract away some of the boilerplate involved in delaying the action of asynchronous methods until [=prerendering browsing context/activated|activation=], we introduce the <dfn extended-attribute>[DelayWhilePrerendering]</dfn> Web IDL extended attribute. It indicates that when a given method is called in a [=prerendering browsing context=], it will immediately return a pending promise and do nothing else. Only upon activation will the usual method steps take place, with their result being used to resolve or reject the previously-returned promise.
+
+The {{[DelayWhilePrerendering]}} extended attribute must take no arguments, and must only appear on a <a lt="regular operation" spec="WEBIDL">regular</a> or <a spec="WEBIDL">static operation</a> whose <a spec="WEBIDL">return type</a> is a <a spec="WEBIDL">promise type</a> and whose <a spec="WEBIDL">exposure set</a> contains {{Window}}.
```

Comment:
Eventually we will probably want to support Workers/Worklets but it will need some sort of recursive definition that's more complex than Chromium has to deal with because according to the spec workers can spawn workers. Something like:
- a service worker is prerendering if all its clients are prerendering
- a shared worker is prerendering if all windows on the origin are prerendering
- a dedicated worker is prerendering if its creator is prerendering
- a worklet is prerendering if its creator is prerendering

and hopefully that eventually reaches the base case of prerendering browsing context, but I'm not sure yet. We can make this a TODO/issue.

Reply:
Might we we eventually want it on promise properties, too? Or would we rather handle that explicitly?

Reply:
There aren't many promise properties on the platform, so my plan would be to leave it out until we find one. (The only ones I know of are in streams.)